client/client_pool/src/external_client.cpp

```diff
@@ -177,25 +177,41 @@ std::pair<int32_t, ConcordClient::PendingReplies> ConcordClient::SendPendingRequ
     request_queue.push_back(bft::client::WriteRequest{single_config, std::move(req.request)});
   }
   try {
-    auto res = new_client_->sendBatch(request_queue, batch_cid);
-    for (const auto& rep : res) {
-      auto cid = seq_num_to_cid.find(rep.first)->second;
-      auto data_size = rep.second.matched_data.size();
-      for (auto& reply : pending_replies_) {
-        if (reply.cid != cid) continue;
-        if (reply.cb) {
-          auto result = bftEngine::SendResult{rep.second};
-          reply.cb(std::move(result));
-          LOG_DEBUG(logger_, "Request processing via callback completed " << KVLOG(client_id_, batch_cid, reply.cid));
+    LOG_INFO(logger_, "Batch processing started" << KVLOG(client_id_, batch_cid));
+    auto received_replies_map = new_client_->sendBatch(request_queue, batch_cid);
+    for (const auto& received_reply_entry : received_replies_map) {
+      const auto received_reply_seq_num = received_reply_entry.first;
+      const auto& pending_seq_num_to_cid_entry = seq_num_to_cid.find(received_reply_seq_num);
+      if (pending_seq_num_to_cid_entry == seq_num_to_cid.end()) {
```

Comment:
Thanks for the check @yuliasherman 
Let's not call find two times on map. 
auto cid = pending_seq_num_to_cid_entry->second